service/history/events/cache.go

```diff
@@ -136,11 +159,7 @@ func newEventKey(domainID, workflowID, runID string, eventID int64) eventKey {
 	}
 }
 
-func (e *cacheImpl) GetEvent(
-	domainID, workflowID, runID string,
-	firstEventID, eventID int64,
-	branchToken []byte,
-) (*shared.HistoryEvent, error) {
+func (e *cacheImpl) GetEvent(shardID int, domainID string, workflowID string, runID string, firstEventID int64, eventID int64, branchToken []byte) (*shared.HistoryEvent, error) {
```

Comment:
As you have the workflowID, can we have the number of shards when initializing the cache, and just use the workflowID and the number of shards to calculate the shard ID?

Reply:
nice idea but it may add some latency here though